### PR TITLE
Use NibbleKey for insert_leaf

### PR DESCRIPTION
**Pull request by gballet**
This is to make sure that the passed data is always in the correct format.

### CODE REVIEW

**Comment by axic** [comment]
Need to also update README.

**Comment by gballet** [comment]
I have just added some code so that the leaf's value is no longer hardcoded in `make_tree_from_json`, and instead is generated for the JSON itself. This is to be expanded into a rewrite of the deserializer.

File: src/utils.rs [hunk 1]
```diff
@@ -56,6 +62,26 @@ impl NibbleKey {
     pub fn len(&self) -> usize {
         self.0.len()
     }
+
+    pub fn with_hex_prefix(&self, is_terminator: bool) -> Vec<u8> {
```
Comment (s1na) [inline src/utils.rs:68]: // This method is adding the hex prefix AND converting the nibble key to a byte key.

**Review by s1na — APPROVED** [review]
Looks good. Later on two additional methods could be added for ease of use: an `insert` which takes a `ByteKey` (or vec) and converts to `NibbleKey` and calls `insert_leaf`. A `insert_secure` which does the same, but also hashes the key.

**Pull request merged**